nasdaqdatalink/api_config.py

```diff
@@ -17,6 +17,18 @@ class ApiConfig:
     retry_status_codes = [429] + list(range(500, 512))
     verify_ssl = True
 
+    def read_key(self, filename=None):
```

Comment:
This function should be called `read_key_from_file`. I was thinking it would be nice to allow each ApiConfig instances to read keys from different files.

Reply:
Ah, but even so it still duplicates the namespace method, which I'd like to avoid having multiple methods that do the same thing.  Ideally the behaviour for finding a key is to:

- check if the env var exists, and return what it is so we can set it; 
- otherwise, check if a file (user defined or default) exists and read from it: if a value is provided set it; 
- we couldn't detect an api key so we must fail

I didn't catch any logic that attempted to call ApiConfig.read_key() or what you propose should be ApiConfig.read_key_from_file() -- so I'm curious whether that's the correct level to provide those helper methods.

ApiConfig is just being used as a [dataclass](https://docs.python.org/3/library/dataclasses.html) anyway, and at some point in the future we should convert it to one. We can have helper methods in it, for sure, but we need to make sure that it should be responsible for doing those tasks.

Reply:
i was think something like this:
```python
api_config1 = ApiConfig()
# api_config1 will have whatever is inside the ApiConfig class
ndl.get('table', api_config=api_config1)
api_config2 = ApiConfig()
api_config2.read_key_from_file('test2.txt')
ndl.get('table', api_config=api_config2)
```
Basically, i was trying to avoid using ApiConfig as a mutable storage of global state, but I understand it does introduce additional complexity to the users.